tests/test_range.py

```diff
@@ -789,6 +789,89 @@ def test_week_start(self):
 
+    def test_weekend(self):
+        now = datetime.now()
+
+        # Frozen date is on a weekend
+        self.assert_range('this weekend',
+                          datetime(2017, 6, 16, WEEKEND_START_HOUR),
+                          datetime(2017, 6, 19, WEEKEND_END_HOUR))
+
+        self.assert_range('this weekend',
+                          datetime(2017, 6, 16, WEEKEND_START_HOUR),
+                          now,
+                          context=CONTEXT_PAST)
+
+        self.assert_range('this weekend',
+                          now,
+                          datetime(2017, 6, 19),
+                          context=CONTEXT_FUTURE)
+
+        self.assert_range('last weekend',
+                          datetime(2017, 6, 9, WEEKEND_START_HOUR),
+                          datetime(2017, 6, 12, WEEKEND_END_HOUR))
+
+        self.assert_range('next weekend',
+                          datetime(2017, 6, 23, WEEKEND_START_HOUR),
+                          datetime(2017, 6, 26, WEEKEND_END_HOUR))
+
+        self.assert_range('since this weekend',
+                          datetime(2017, 6, 16, WEEKEND_START_HOUR),
+                          now)
+
+        self.assert_range('since last weekend',
+                          datetime(2017, 6, 9, WEEKEND_START_HOUR),
+                          now)
+
+        self.assert_range('until next weekend',
+                          now,
+                          datetime(2017, 6, 23, WEEKEND_START_HOUR))
+
+        # On a non-weekend
+        with freeze_time('2017-06-14 10:43:09'):
+            self.assert_range('this weekend',
+                              datetime(2017, 6, 16, WEEKEND_START_HOUR),
+                              datetime(2017, 6, 19, WEEKEND_END_HOUR))
+
+            self.assert_range('this weekend',
+                              datetime(2017, 6, 16, WEEKEND_START_HOUR),
+                              datetime(2017, 6, 19, WEEKEND_END_HOUR),
+                              context=CONTEXT_PAST)
+
+            self.assert_range('this weekend',
+                              datetime(2017, 6, 16, WEEKEND_START_HOUR),
+                              datetime(2017, 6, 19, WEEKEND_END_HOUR),
+                              context=CONTEXT_FUTURE)
+
+            self.assert_range('last weekend',
+                              datetime(2017, 6, 9, WEEKEND_START_HOUR),
+                              datetime(2017, 6, 12, WEEKEND_END_HOUR))
+
+            self.assert_range('next weekend',
+                              datetime(2017, 6, 16, WEEKEND_START_HOUR),
+                              datetime(2017, 6, 19, WEEKEND_END_HOUR))
+
+            self.assert_range('since last weekend',
+                              datetime(2017, 6, 9, WEEKEND_START_HOUR),
+                              datetime.now())
+
+            self.assert_range('until next weekend',
+                              datetime.now(),
+                              datetime(2017, 6, 16, WEEKEND_START_HOUR))
```

Comment:
next weekend is still the weekend of the next week if you are on a weekday  datetime(2017, 6, 23, WEEKEND_START_HOUR)